types/index.d.ts

```diff
@@ -157,9 +157,9 @@ declare namespace unified {
     plugins?: PluggableList
     settings?: Settings
   }
-  type PluginTuple = [Plugin, Settings]
-  type Pluggable = Plugin | Preset | PluginTuple
-  type PluggableList = Pluggable[]
+  type PluginTuple<T = Settings> = [Plugin<T>, T]
+  type Pluggable<T = Settings> = Plugin<T> | Preset | PluginTuple<T>
+  type PluggableList<T = Settings> = Array<Pluggable<T>>
```

Comment:
There is an edge case here.
If two plugins declare different interfaces there will be a clash.

For example:

```ts
interface ExamplePluginSettings {
  example: string
}
interface ExamplePluginSettings2 {
  example2: string
}

const typedPlugin: Plugin<ExamplePluginSettings> = function() {}
const typedPlugin2: Plugin<ExamplePluginSettings2> = function() {}

const typedSetting = {example: 'example'}
const typedSetting2 = {example2: 'example'}

processor.use([[typedPlugin, typedSetting], [typedPlugin2, typedSetting2]])
```

will result in

```
Type '(Attacher<ExamplePluginSettings2> | { example2: string; })[]' is not assignable to type 'Pluggable<{ example: string; }>'.
  Type '(Attacher<ExamplePluginSettings2> | { example2: string; })[]' is missing the following properties from type '[Attacher<{ example: string; }>, { example: string; }]': 0, 1ts(2322)
```

There are a few options here:

1. set the generic to `unknown` or `any` and lose the strong typing in this case
2. leave the typing as is, typescripters can work around the issue by chaining `use` methods, one per typed plugin
3. something else?

Reply:
> set the generic to unknown or any and lose the strong typing in this case

I think this is the best.

Reply:
Updated to use `any`